### PR TITLE
Test suites for OnlineLoader in kes-dashboard-live-tiles

### PR DESCRIPTION
**Pull request by radujipa**
Together with any bug fixing and refactors, this includes tests
for the OnlineLoader in the KES module.

### CODE REVIEW

File: test/fixtures/kes_dlt_cli/KesDltResponsesFixture.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        for (const auto &keyValue : responses) {
+            if (keyValue.second == nullptr) {
+                cerr << "KesDltResponsesFixture: FAILED TO LOAD DATA FOR "
+                     << keyValue.first << endl;
+            }
+        }
+
+        for (const auto &keyValue : responses) {
+            tileCounts[keyValue.first] =
+                KesDltResponsesFixture::getTileCount(keyValue.second);
```
Comment (tombettany) [inline test/fixtures/kes_dlt_cli/KesDltResponsesFixture.h]: Could be merged with the loop immediately prior.

File: test/kes_dashboard_live_tiles_cli/TestOnlineLoader.h [hunk 2]
```diff
@@ -12,24 +12,265 @@
...
+ * Check that for a given HTTPClient.GET() response OnlineLoader.getTiles()
+ * calls for each tile data Tile.initialise() with the expected JSON.
+ */
+TEST_P(KesDltValidResponsesFixture, GetTilesCallsTileInitialise) {
+    shared_ptr<MockHTTPClient> mockHttpClient = make_shared<MockHTTPClient>();
+    shared_ptr<MockTileFactory> mockTileFactory = make_shared<MockTileFactory>();  // NOLINT
+    shared_ptr<MockTile> mockTile = make_shared<MockTile>();
+
+    OnlineLoader onlineLoader(
+        "/tmp/kes-dlt-cli/test", mockHttpClient, mockTileFactory);
```
Comment (tombettany) [inline test/kes_dashboard_live_tiles_cli/TestOnlineLoader.h]: Are you cleaning up this directory at the end? Also, maybe I've missed it but where is this directory being created?

File: test/kes_dashboard_live_tiles_cli/TestOnlineLoader.h [hunk 3]
```diff
@@ -12,24 +12,265 @@
...
+}
+
+/**
+ * Check that for a given HTTPClient.GET() response OnlineLoader.getTiles()
+ * calls for each tile data Tile.download() with the given cacheDir.
+ */
+TEST_P(KesDltValidResponsesFixture, GetTilesCallsTileDownload) {
+    shared_ptr<MockHTTPClient> mockHttpClient = make_shared<MockHTTPClient>();
+    shared_ptr<MockTileFactory> mockTileFactory = make_shared<MockTileFactory>();  // NOLINT
+    shared_ptr<MockTile> mockTile = make_shared<MockTile>();
```
Comment (tombettany) [inline test/kes_dashboard_live_tiles_cli/TestOnlineLoader.h]: I think that this is one of the few cases where you might actually want to use `auto`.

File: test/kes_dashboard_live_tiles_cli/TestOnlineLoader.h (hunk 2 above)
Comment (radujipa) [reply test/kes_dashboard_live_tiles_cli/TestOnlineLoader.h]: Cleaning up manually right now, need to sort it out - just haven't come up with a clean way of doing it yet 🤔 Directories right now are being created in Tile.download(). 

Actually, maybe ITileLoader should include something like `createCache`, `clearCache`.

File: include/mercury/kes_dashboard_live_tiles_cli/ILocalLoader.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+#include <memory>
+
+#include "mercury/kes_dashboard_live_tiles_cli/ITile.h"
+#include "mercury/kes_dashboard_live_tiles_cli/ITileLoader.h"
+
+using std::list;
+using std::shared_ptr;
+
+
+class ILocalLoader : public ITileLoader {
```
[the remark this answers is not in the document]
Comment (radujipa) [reply include/mercury/kes_dashboard_live_tiles_cli/ILocalLoader.h]: Thinking of keeping Online and Local Loaders internal to the KES and only expose `TileManager` to users. Will work out my namespaces in the future.

**Pull request merged**